### DISCUSSION TITLE
Welcome to QE Book Discussions!

### DISCUSSION THREAD

**Discussion by nguyen-group** [nguyen-group/QE-SSP#2] [Announcements]
Dear Readers,

Thank you for using our book and welcome to the discussion.
We’re using Discussions as a place to connect with the readers. We hope that you:
- Ask questions about the contents of the book, such as difficulty in installation, etc.
- Share your impressions about the book.
- Report the error or bug of the input files. 

Any suggestions to improve the book are welcome.

On behalf of the authors!

**Comment by r-saito-sendai** [comment]
Hi. I, Saito is a co-author of the book. Thank you for your interest on our book. Any questions or comments are welcome.

**Comment by gabriel-2023** [comment]
Hello Nguyen and Saito, I want to learn how to use Quantum Espresso and I am following your book to install it. Unfortunately, I have run into problems while following your installation procedure. 
1) when I type jupyter-lab it says No web browser found: Error('could not locate runnable browser').
2) when I type mpirun -np 4 pw.x < scf.in > scf.out & I get the following message 
Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG IEEE_DENORMAL
Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG IEEE_DENORMAL
Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG IEEE_DENORMAL
Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG IEEE_DENORMAL

3) When I type xcrysden & it is displayed like in the following picture
![Captura de pantalla 2023-12-11 201717](https://github.com/nguyen-group/QE-SSP/assets/153573889/e3af3af2-8d4e-421d-ad3d-18b354608d8c)

![Captura de pantalla 2023-12-11 201747](https://github.com/nguyen-group/QE-SSP/assets/153573889/cd2fef0d-1ac7-41db-a5eb-00fa40cf0cf0)

Can you help me out to install correctly Quantum Espresso please.
Regards
Gabriel

**Reply by nguyen-group** [reply]
Hi,

1) when I type jupyter-lab it says No web browser found: Error('could not locate runnable browser').

Can you see something like a ULR? http://localhost:8888/?token=d0a958999d7e59da3f097809882e8f40eb5845e3a07cebe8
you can copy and paste it into your web browser.

2) Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG IEEE_DENORMAL
This is not an error but a message. I think pw.x program can still run normally. 

3) When I type xcrysden & it is displayed like in the following picture
I never see it, so I have no idea yet. Another choice is to reinstall everything. 

By the way, which Linux OS that you install in Windows?
If you want to use the Windows, I suggest the install Ubuntu on WSL2 (https://ubuntu.com/tutorials/install-ubuntu-on-wsl2-on-windows-11-with-gui-support#1-overview)

**Comment by gabriel-2023** [comment]
Thank you very much for your response.

I have reinstalled everything and now it is working fine.

Thanks again

Regards
Gabriel
________________________________
De: Quantum Materials @ Tohoku University ***@***.***>
Enviado: martes, 12 de diciembre de 2023 15:28
Para: nguyen-group/QE-SSP ***@***.***>
Cc: GABRIEL GONZALEZ CONTRERAS ***@***.***>; Comment ***@***.***>
Asunto: Re: [nguyen-group/QE-SSP] Welcome to QE Book Discussions! (Discussion #2)


Hi,

  1.  when I type jupyter-lab it says No web browser found: Error('could not locate runnable browser').

Can you see something like a ULR? http://localhost:8888/?token=d0a958999d7e59da3f097809882e8f40eb5845e3a07cebe8
you can copy and paste it into your web browser.

  1.  Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG IEEE_DENORMAL
This is not an error but a message. I think pw.x program can still run normally.

  2.  When I type xcrysden & it is displayed like in the following picture
I never see it, so I have no idea yet. Another choice is to reinstall everything.

By the way, which Linux OS that you install in Windows?
If you want to use the Windows, I suggest the install Ubuntu on WSL2 (https://ubuntu.com/tutorials/install-ubuntu-on-wsl2-on-windows-11-with-gui-support#1-overview)

**Comment by AlbertFernandoF** [comment]
Hi,  

I’m currently studying the derivation of the Kohn-Sham equations in Chapter 4 and have some difficulties with Equation 4.62:  

1. I’m unclear about the origin of the -3Z term. The text mentions it comes from 2 \epsilon, but isn’t \epsilon = -Z^2 /2? Also, i feel like the integral in this equation is challenging to evaluate analytically. Could you also clarify the source of the 2\pi and r^2 factors?  
2. When solving dE/dZ = 0, I find the resulting roots do not align with Z = 1.8. Instead, I get Z = 1.25.  
3. Substituting Z = 1.8 back into Equation 4.62 does not yield E = -2.7. Instead, I get E = -1.55.  

Could you please provide clarification or point me to additional resources to help resolve these issues?

**Reply by nguyen-group** [reply]
Hi,

Q1: To analyze the integral in Equation 4.62, we need to use integrals in spherical coordinates. Please find the details how to convert the integrals in https://github.com/nguyen-group/QE-SSP/discussions/3, then you will see why we have the 2\pi and r^2 factor. I noted that we have a typo in Equation 4.62, it should be E = Z^2-3Z-Z^3*(Z+4)/(Z+2)^3.

Q2 and Q3: Please check with the corrected equation 4.62: E = Z^2-3Z-Z^3*(Z+4)/(Z+2)^3. I believe that Z=1.826 is a solution. 
I am sorry that I lost my drafts and was lazy, so I used chatGPT to check.
<img width="395" alt="Screenshot 2024-11-26 at 0 36 29" src="https://github.com/user-attachments/assets/51346856-a243-482b-9693-08a494e056e2">

Nguyen

**Reply by AlbertFernandoF** [reply]
What about the 2\epsilon = Z^2 - 4Z? i still can't grasp where this idea comes from. isnt \epsilon = -Z^2/2?

**Reply by nguyen-group** [reply]
> What about the 2\epsilon = Z^2 - 4Z? i still can't grasp where this idea comes from. isnt \epsilon = -Z^2/2?

\epsilon = -Z^2/2 is energy without Hartree potential V_H, as defined in Eq. 4.10: h(r)\phi(r)= \epsilon\phi(r).
However, in the case of a helium atom, we consider V_H. Thus,  \epsilon is defined in Eq. 4.48: [h(r)+V_H]\phi(r)= \epsilon\phi(r).
After several steps of doing the integrals, we can get  \epsilon = Z^2/2-2Z.
